public/scripts/routes.js

```diff
@@ -0,0 +1,6 @@
+
+page('/');
+page('/coffee', coffeePageController.index);
```

Comment:
This is a 'Think it over' comment...

You know that page.js can handle additional callbacks, which may be another way of solving the async issue that Becky helped you with. I didn't think of it until this week, but it would give you the ability to make a route request, call one callback / ajax request, and then when done, call the `next()`. Go back and take a look at how we used this in the blog assignment, and see if it might improve the code fluidity that you've already written.